### PR TITLE
Add localization cli command

### LINKED ISSUE #9708: Use a translation service for Theia's own translations
### Feature Description:

With #9538 we will add general i18n support for Theia. While most user facing content can be automatically translated using the vscode language packs, text that only exists in Theia won't be translated so easily. I don't think translating each contributed user facing string into 10+ languages is feasible for Theia on each PR. Therefore, we need to think of a solution how to translate Theia's own content.

Microsoft is using [`transifex`](https://www.transifex.com/) combined with the [`vscode-nls-dev`](https://github.com/Microsoft/vscode-nls-dev) package to generate json files from vscode's source which can then be machine translated into different languages.

My proposal would be to mimic the `vscode-nls-dev` behavior in a dev-package to create json files which be translated on each release.

### PR DESCRIPTION
**Pull request by msujew**
#### What it does
Closes #9708

Uses the machine translation API of DeepL to automatically translate json files.

- [x] [CQ for DeepL usage](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23832)

#### How to test

Testing this PR requires a DeepL API key. While 500k characters/month are free for developers, even a free account requires a credit card (claiming to reduce the amount of multi-accounts that abuse the free dev-tier). If you want to review this PR but don't have access to an account already, or are hesitant to use your own credit card there, you can send me a mail (linked in my github profile) and I'll send you an API key.

1. Create a sample file that you want to translate:
```json
{
  "test": "This is a value",
  "nested": {
    "hi": "Hello",
    "deeper": {
      "why": "Why"
    }
  }
}
```
2. Run the cli:
```sh
yarn theia localize -f test.json -k <api-key> (--free-api) <language codes like de, it, fr>
```
3. Assert that the output files are correctly translated (they will be in the same directory as the source file)
4. Make use of different arguments, or play around with the `help` of the localize command
5. Change the sample file and assert that only new entries which can't be found in the already translated files are actually translated (the cli will output the amount of translated entries)

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by vince-fugnitto — DISMISSED** [review]
@brianking @marcdumais-work just as a precaution, I'd like your input about using [deepl](https://www.deepl.com/pro-api?cta=header-pro) (which is a free or paid service that requires an api key and credit card) in order to perform translations. Is there anything from the eclipse foundation point of view which discourages doing such things?

**Comment by vince-fugnitto** [comment]
> We don't. The `extract` command will perform all necessary preprocessing to write a simple json file to the file system, which only contains the translation IDs used and their (english) default values. Later, the `localize` command (using deepl) takes that json file (which does not contain any code) and translates the default values into the specified language.

@msujew unfortunately even something as simple as `ids` from proprietary extensions can reference proprietary components and so-on, and I can think of many use-cases where it would not be acceptable by companies :(

File: dev-packages/localization-manager/README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 The `@theia/localization-manager` package is used easily create localizations of Theia and Theia extensions for different languages. 
 Its main use case is to extract localization keys and default values from `nls.localize` calls within the codebase.
+Additionally, it uses the [DeepL API](https://www.deepl.com/docs-api) to automatically translate any missing localization values.
```
Comment (vince-fugnitto) [inline dev-packages/localization-manager/README.md:17]: Should we enhance the documentation for the `DeepL` functionality to describe it further, and potentially include examples? We might want to describe how it works.

**Review by vince-fugnitto — COMMENTED** [review]
@msujew can you describe a bit how the `deepl` integration is supposed to work for theia translations (perhaps in the readme of the extension). It's not instantly clear to me what it does or how it works. Some insights on how translations of theia should work, and if we should maintain some type of theia `nls.metada.json` in the framework for our own translations.

**Comment by msujew** [comment]
@vince-fugnitto I expanded the readme and added an example. I would refrain from adding more documentation in there, instead choosing to dedicate a larger section to it in the [internationalization documentation](https://theia-ide.org/docs/i18n/). Regarding, a `nls.metadata.json` for Theia itself, I would rather document this in the coding guidelines. What do you think about that?

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto I expanded the readme and added an example. I would refrain from adding more documentation in there, instead choosing to dedicate a larger section to it in the [internationalization documentation](https://theia-ide.org/docs/i18n/). Regarding, a `nls.metadata.json` for Theia itself, I would rather document this in the coding guidelines. What do you think about that?

@msujew I'd be fine with whatever you propose :) the important thing is we have some form of documentation to help developers, and have a clear way forward.

**Comment by JonasHelming** [comment]
@vince-fugnitto : Are you fine with merging this?

File: dev-packages/localization-manager/package.json [hunk 2]
```diff
@@ -29,7 +29,10 @@
     "watch": "theiaext watch"
   },
   "dependencies": {
+    "@types/bent": "^7.0.1",
     "@types/fs-extra": "^4.0.2",
+    "bent": "^7.1.0",
+    "colors": "^1.4.0",
```
Comment (vince-fugnitto) [inline dev-packages/localization-manager/package.json:35]: @msujew can we replace colors with `chalk` after https://github.com/eclipse-theia/theia/pull/10612?

Comment (msujew) [reply dev-packages/localization-manager/package.json]: Rebased and replaced `colors` with `chalk` 👍

**Review by vince-fugnitto — APPROVED** [review]
LGTM! @msujew I don't have any further comments regarding the code :+1:

**Comment by msujew** [comment]
@vince-fugnitto Thanks, I'll merge it after the release 👍

**Comment by JonasHelming** [comment]
@msujew : Release is done, "ping" :-)

**Comment by JonasHelming** [comment]
@msujew Can this be merged?

**Pull request merged**